manager/integration/tests/test_infra.py

```diff
@@ -0,0 +1,129 @@
+import os
+import pytest
+import time
+
+from digitalocean import digitalocean
+from common import get_core_api_client, get_longhorn_api_client
+from common import RETRY_COUNTS, RETRY_INTERVAL
+
+
+def detect_cloudprovider():
+    cloudprovider_name = os.getenv("CLOUDPROVIDER")
+
+    if cloudprovider_name == "digitalocean":
+        cloudprovider = digitalocean()
+        return cloudprovider
+
+
+def is_node_ready_k8s(node_name):
+    k8s_api_client = get_core_api_client()
+
+    node_status_k8s = k8s_api_client.read_node_status(node_name)
+
+    for node_condition in node_status_k8s.status.conditions:
+        if node_condition.type == "Ready" and \
+           node_condition.status == "True":
+            node_ready = True
+        else:
+            node_ready = False
+
+    return node_ready
+
+
+def is_node_ready_longhorn(node_name):
+    longhorn_api_client = get_longhorn_api_client()
+
+    for longhorn_node in longhorn_api_client.list_node():
+        if longhorn_node['name'] == node_name:
+            node = longhorn_node
+            break
+
+    if node.conditions.Ready.status == 'False':
+        node_ready = False
+    else:
+        node_ready = True
+
+    return node_ready
+
+
+@pytest.yield_fixture
+def reset_cluster_ready_status():
+    yield
+    node_worker_label = 'node-role.kubernetes.io/worker'
+
+    k8s_api_client = get_core_api_client()
+    cloudprovider = detect_cloudprovider()
+
+    for node_item in k8s_api_client.list_node().items:
+        if node_worker_label in node_item.metadata.labels and \
+                node_item.metadata.labels[node_worker_label] == 'true':
+            node_name = node_item.metadata.name
+
+            if is_node_ready_k8s(node_name) is False:
+                node = cloudprovider.node_id(node_name)
+
+                cloudprovider.node_start(node)
+
+                for i in range(RETRY_COUNTS):
```

Comment:
We can implement `wait_for_node_up()` here.

Reply:
done